src/internal/m365/exchange/contacts_restore_test.go

```diff
@@ -175,6 +175,19 @@ func (suite *ContactsRestoreIntgSuite) TestRestoreContact() {
 				assert.True(t, m.calledDelete, "old item deleted")
 			},
 		},
+		{
+			name:         "collision: replace - err already deleted",
+			apiMock:      &contactRestoreMock{deleteItemErr: graph.ErrDeletedInFlight},
+			collisionMap: map[string]string{collisionKey: "smarf"},
+			onCollision:  control.Replace,
+			expectErr: func(t *testing.T, err error) {
+				assert.NoError(t, err, clues.ToCore(err))
+			},
+			expectMock: func(t *testing.T, m *contactRestoreMock) {
+				assert.True(t, m.calledPost, "new item posted")
+				assert.True(t, m.calledDelete, "old item deleted")
```

Comment:
this should be checking the ID of the deleted item (same with other data types too)

Reply:
good catch, thanks.